tfx/examples/cifar10/cifar10_utils_native_keras.py

```diff
@@ -0,0 +1,338 @@
+  # TODO(b/157064428): Support label transformation for Keras.
+  # Do not apply label transformation as it will result in wrong evaluation.
+  outputs[transformed_name(LABEL_KEY)] = inputs[LABEL_KEY]
+
+  return outputs
+
+def _write_metadata(model_path: str,
+                    label_map_path: str,
+                    mean: list,
+                    std: list):
+  """ Add TFLite metadata to the model
+
+  Args:
+    model_path: The path of the TFLite model
+    label_map_path: The path of the label map file
+    mean: The mean value used to normalize input image tensor
+    std: The standard deviation used to normalize input image tensor
+  """
+
+  model_meta = _metadata_fb.ModelMetadataT()
+
+  # Creates input normalization info.
+  input_meta = _metadata_fb.TensorMetadataT()
+  input_normalization = _metadata_fb.ProcessUnitT()
+  input_normalization.optionsType = (
+      _metadata_fb.ProcessUnitOptions.NormalizationOptions)
+  input_normalization.options = _metadata_fb.NormalizationOptionsT()
+  input_normalization.options.mean = mean
+  input_normalization.options.std = std
+  input_meta.processUnits = [input_normalization]
+
+  # Creates output label map info.
+  output_meta = _metadata_fb.TensorMetadataT()
+  label_file = _metadata_fb.AssociatedFileT()
+  label_file.name = os.path.basename(label_map_path)
+  label_file.type = _metadata_fb.AssociatedFileType.TENSOR_AXIS_LABELS
+  output_meta.associatedFiles = [label_file]
+
+  # Creates subgraph info.
+  subgraph = _metadata_fb.SubGraphMetadataT()
+  subgraph.inputTensorMetadata = [input_meta]
+  subgraph.outputTensorMetadata = [output_meta]
+  model_meta.subgraphMetadata = [subgraph]
+
+  b = flatbuffers.Builder(0)
+  b.Finish(
+      model_meta.Pack(b),
+      _metadata.MetadataPopulator.METADATA_FILE_IDENTIFIER)
+  metadata_buf = b.Output()
+
+  # Populates metadata and label file to the model file.
+  populator = _metadata.MetadataPopulator.with_model_file(model_path)
+  populator.load_metadata_buffer(metadata_buf)
+  populator.load_associated_files([label_map_path])
+  populator.populate()
+
+
+# TFX Trainer will call this function.
+def run_fn(fn_args: TrainerFnArgs):
+  """Train the model based on given args.
+
+  Args:
+    fn_args: Holds args used to train the model as name/value pairs.
+  """
+  tf_transform_output = tft.TFTransformOutput(fn_args.transform_output)
+  train_dataset = _input_fn(fn_args.train_files, tf_transform_output, True,
+                            _TRAIN_BATCH_SIZE)
+  eval_dataset = _input_fn(fn_args.eval_files, tf_transform_output, False,
+                           _EVAL_BATCH_SIZE)
+
+  mirrored_strategy = tf.distribute.MirroredStrategy()
+  with mirrored_strategy.scope():
+    model, base_model = _build_keras_model()
+
+  # Our training regime has two phases: we first freeze the backbone and train
+  # the newly added classifier only, then unfreeze part of the backbone and
+  # fine-tune with classifier jointly.
```

Comment:
Lets make it clear in the comment that we evenly share the total number of train_steps between the two phases.